### PR TITLE
(experimental) refactor: [ANDROSDK-1720] refactor Call classes to Kotlin

### PR DESCRIPTION
**Pull request by dan-dhis2**
From PR [2019](https://github.com/dhis2/dhis2-android-sdk/pull/2019)

_Refactor the following

UidsCall
QueryCall
ListCall
ObjectCall
And their implementations, to Kotlin_

### CODE REVIEW

**Review by vgarciabnz — CHANGES_REQUESTED** [review]
It looks good. Two general comments that can be done in the next PR:
- Remove "java.lang.Boolean" dependencies. They were added in the conversion to Kotlin and should be replaced by native Kotlin boolean.
- Most of the call classes can be internal

File: core/src/main/java/org/hisp/dhis/android/core/program/internal/ProgramCall.kt [hunk 1]
```diff
@@ -25,48 +25,49 @@
...
 @Reusable
-public final class ProgramCall implements UidsCall<Program> {
-
-    private static final int MAX_UID_LIST_SIZE = 50;
+class ProgramCall @Inject internal constructor(
+    private val service: ProgramService,
+    handler: ProgramHandler,
+    apiDownloader: APIDownloader,
+) : UidsCall<Program> {
+    private val handler: Handler<Program>
```
[the remark this answers is not in the document]
Comment (dan-dhis2) [reply core/src/main/java/org/hisp/dhis/android/core/program/internal/ProgramCall.kt:46]: To be included in RxJava refactor in MetadataCall [Jira Issue ANDROSDK-1683](https://dhis2.atlassian.net/browse/ANDROSDK-1683)

File: core/src/main/java/org/hisp/dhis/android/core/event/internal/EventFilterCall.kt [hunk 2]
```diff
@@ -25,60 +25,59 @@
...
+import org.hisp.dhis.android.core.common.internal.DataAccessFields
+import org.hisp.dhis.android.core.event.EventFilter
+import org.hisp.dhis.android.core.systeminfo.DHISVersion
+import org.hisp.dhis.android.core.systeminfo.DHISVersionManager
+import java.lang.Boolean
+import javax.inject.Inject
 
 @Reusable
-public final class EventFilterCall implements UidsCall<EventFilter> {
+class EventFilterCall @Inject internal constructor(
```
[the remark this answers is not in the document]
Comment (dan-dhis2) [reply core/src/main/java/org/hisp/dhis/android/core/event/internal/EventFilterCall.kt:44]: To be included in RxJava refactor in MetadataCall [Jira Issue ANDROSDK-1683](https://dhis2.atlassian.net/browse/ANDROSDK-1683)

File: core/src/main/java/org/hisp/dhis/android/core/event/internal/EventFilterCall.kt [hunk 3]
```diff
@@ -25,60 +25,59 @@
...
+    private val service: EventFilterService,
+    handler: EventFilterHandler,
+    apiDownloader: APIDownloader,
+    versionManager: DHISVersionManager,
+) : UidsCall<EventFilter> {
+    private val handler: Handler<EventFilter>
```
[the remark this answers is not in the document]
Comment (dan-dhis2) [reply core/src/main/java/org/hisp/dhis/android/core/event/internal/EventFilterCall.kt:50]: To be included in RxJava refactor in MetadataCall [Jira Issue ANDROSDK-1683](https://dhis2.atlassian.net/browse/ANDROSDK-1683)

**Comment by dan-dhis2** [comment]
To be closed by @vgarciabnz 
As all requested changes will be included in RxJava refactor in MetadataCall [Jira Issue ANDROSDK-1683](https://dhis2.atlassian.net/browse/ANDROSDK-1683)

**Review by vgarciabnz — APPROVED** [review]

**Pull request merged**